### PR TITLE
Added gesture Length to Push and Thrust to avoid false detection

### PR DESCRIPTION
**Pull request by fdarosa2663**
Per your suggestion I've added Gesture Length (ms) to both Push and Thrust to avoid false detections.  I've also included separate defines for each to allow the user to modify as needed in the config.  We'll need the users who experienced the false detections to test as none of my sabers experienced the false detections.

### CODE REVIEW

File: props/saber_fett263_buttons.h [hunk 1]
```diff
@@ -110,10 +110,16 @@
 // FETT263_THRUST_ON_NO_BM
 // To enable Thrust On Ignition control but not activate Battle Mode (works with THRUST_ON_PREON only)
 //
+// FETT263_THRUST_LENGTH 50
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:113]: I don't really think you need defines for everything.
You don't really need to change it, but too many defines are just going to confuse users and make it harder to keep things backwards compatible. People who really want to tweak things can just fiddle with the code directly.

File: props/saber_fett263_buttons.h [hunk 2]
```diff
@@ -251,10 +265,18 @@ SaberFett263Buttons() : PropBase() {}
 
       // EVENT_PUSH
       if (fabs(mss.x) < 3.0 &&
-          mss.y * mss.y + mss.z * mss.z > 120 &&
+          mss.y * mss.y + mss.z * mss.z > 100 &&
           fusor.swing_speed() < 30 &&
           fabs(fusor.gyro().x) < 10) {
-        Event(BUTTON_NONE, EVENT_PUSH);
+          if (!push_begin_) {
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:271]: I don't think this works that well.
Basically, when pushing is detected, it starts a timer, and when that timer expires, the next time pushing is detected, a push event is generated.

I think what you want is something like this:

    if ( DETECT PUSH ) {
       if (millis() - push_begin_millis_ > FETT263_PUSH_LENGTH) {
         Event(BUTTON_NONE, EVENT_PUSH);
         push_begin_millis_ = millis();
       }
     } else {
        push_begin_millis_ = millis();
     }

Basically, push_begin_millis_ is set to millis every time a push is NOT detected. That way, millis() - push_begin_millis_ will only become 25 if push was detected every time we get to this code for 25 milliseconds.

Same thing goes for thrust below.

File: props/saber_fett263_buttons.h [hunk 3]
```diff
@@ -411,12 +441,18 @@ SaberFett263Buttons() : PropBase() {}
...
           saber_off_time_ = millis();
+        }
+#else
+         Off();
+         saber_off_time_ = millis();
+         battle_mode_ = false;
 #endif
-#ifndef FETT263_BATTLE_MODE_ALWAYS_ON
-          battle_mode_ = false;
+#ifdef FETT263_BATTLE_MODE_ALWAYS_ON
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:454]: Did you mean to change this back again?
Which is it supposed to be?

Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:454]: this was tied to the previous change that was merged to handle the button changing to false.

File: props/saber_fett263_buttons.h (hunk 2 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:271]: thanks, I was thinking the push_begin_millis_ would be continuously overwritten if I didn't include a bool, I've updated both the same, it's working for me but the target for this is the users with the speaker vibrations so will need their testing to confirm

File: props/saber_fett263_buttons.h (hunk 1 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:113]: OK, wasn't sure if it made it easier or harder but I've removed.

File: props/saber_fett263_buttons.h [hunk 4]
```diff
@@ -251,10 +251,15 @@ SaberFett263Buttons() : PropBase() {}
 
       // EVENT_PUSH
       if (fabs(mss.x) < 3.0 &&
-          mss.y * mss.y + mss.z * mss.z > 120 &&
+          mss.y * mss.y + mss.z * mss.z > 100 &&
           fusor.swing_speed() < 30 &&
           fabs(fusor.gyro().x) < 10) {
-        Event(BUTTON_NONE, EVENT_PUSH);
+        if (millis() - push_begin_millis_ > 25) {
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:257]: This is not what I suggested, the else is in the wrong place.
As written, this will do roughly the opposite to what we want.
push_begin_millis_ will be set to millis() every time that the push detection triggers.
push events will be generated if it's been more than 25 ms since the last push detection, but it will not do anything to filter out very short events.

Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:257]: sorry, fixed. did have to lower for push as it's a very short movement as opposed to thrust

File: props/saber_fett263_buttons.h [hunk 5]
```diff
@@ -251,10 +251,15 @@ SaberFett263Buttons() : PropBase() {}
 
       // EVENT_PUSH
       if (fabs(mss.x) < 3.0 &&
-          mss.y * mss.y + mss.z * mss.z > 120 &&
+          mss.y * mss.y + mss.z * mss.z > 100 &&
           fusor.swing_speed() < 30 &&
           fabs(fusor.gyro().x) < 10) {
-        Event(BUTTON_NONE, EVENT_PUSH);
+        if (millis() - push_begin_millis_ > 10) {
```
Comment (profezzorn) [inline props/saber_fett263_buttons.h:257]: At 10 ms, this loop will only execute a few times, and it might not be worth doing this extra step.
Wish we could test for "speed" instead of "mss", that would make all of this much better.

**Comment by profezzorn** [comment]
This is ready to merge, but git says it has conflicts that must be resolved first.

**Comment by fdarosa2663** [comment]
How do I resolve conflicts, it's not giving me an option

File: props/saber_fett263_buttons.h (hunk 5 above)
Comment (fdarosa2663) [reply props/saber_fett263_buttons.h:257]: Did we try speed?  It's working for me currently but I can test another method, with the 25 it was inconsistent because the gesture is meant to be a short movement so I had to lower to have it work.  but if we can improve I'm open to trying other options.

**Comment by fdarosa2663** [comment]
looks like indents were the conflicts, should be ok unless you want to test out some other changes to Push first, I'm open to suggestions
